main.py

```diff
@@ -25,93 +25,32 @@ def __init__(self):
+            .triggered.connect(lambda: export_to_clipboard(self))
         self.ui.findChild(QtCore.QObject, ADD_NEW_STREAM) \
-            .triggered.connect(self.add_new_stream)
+            .triggered.connect(lambda: add_new_stream(self))
 
         self.ui.show()
 
-    def mute_all_streams(self):
-        """Toggles the audio of all the players."""
-        for videoframe in self.grid.videoframes:
-            if self.actions[MUTE_CHECKBOX].isChecked():
-                videoframe.player.audio_set_mute(True)
-            else:
-                if not videoframe.is_muted:
-                    videoframe.player.audio_set_mute(False)
-
-    def export_streams_to_clipboard(self):
-        """Exports all streams to the users clipboard."""
-        stream_urls = []
-
-        for videoframe in self.grid.videoframes:
-            stream_urls.append(videoframe.stream.url)
-
-        text = "\n".join(stream_urls)
-
-        clipboard = QtWidgets.QApplication.clipboard()
-        clipboard.clear(mode=clipboard.Clipboard)
-        clipboard.setText(text, mode=clipboard.Clipboard)
-
-    def add_new_stream(self, *args, stream_url=None, stream_quality="best"):
-        """Adds a new player for the specified stream in the grid."""
-        if not stream_url:
-            stream_url, ok = QtWidgets.QInputDialog.getText(self, "Stream input", "Enter the stream URL:")
-
-            if not ok:
-                return
-
-        try:
-            stream_options = self.streamlink_session.streams(stream_url)
-
-            if stream_quality not in stream_options:
-                stream_quality, ok = self._get_user_quality_preference(stream_options)
-
-                if not ok:
-                    return
-
-            self.setup_videoframe(stream_options, stream_quality)
-
-        except streamlink.exceptions.NoPluginError:
-            error_window = QtWidgets.QMessageBox().warning(
-                self,
-                "Error",
-                "Could not open stream: The provided URL is not supported"
-            )
-
-            self.add_new_stream()
-
-    def _get_user_quality_preference(self, stream_options):
-        filtered_qualities = LiveStreamContainer.filtered_quality_options(
-            stream_options
-        )
-
-        return QtWidgets.QInputDialog.getItem(
-            self,
-            "Stream Quality option",
-            """The default stream quality option could not be used.
-            Please select another one:""",
-            reversed(filtered_qualities)
-        )
-
-    def setup_videoframe(self, stream_options, quality):
+    def setup_videoframe(self, stream_url, stream_options, quality):
         """Sets up a videoframe and with the provided stream information."""
-        self.grid.add_new_videoframe(self.vlc_instance, stream_options, quality, self.move_frame)
+        self.grid.add_new_videoframe(self, self.vlc_instance, stream_url, stream_options, quality)
```

Comment:
Since both the functions `self.move_frame` and `self.toggle_fullscreen` is triggered on a `videoframe` level, we have to pass down a reference to those. This feels like really weird design with high coupling and weird referencing, that's really unclear. I'll think about how to refactor this abit and open a new PR on it.